app.py

```diff
@@ -141,6 +133,66 @@ async def create_directory():
         return jsonify(["CreateDirectoryRequestError: " + str(error)]), 400
 
 
+@app.post("/image-validation")
+async def image_validation():
+    """
+    Validates an image based on its extension, header, size, and resizability.
+
+    Returns:
+        A JSON response containing a validator hash.
+
+    Raises:
+        ImageValidationError: If the image fails any of the validation checks.
+    """
+    try:
+
+        data = await request.get_json()
+        image_base64 = data["image"]
+
+        header, encoded_image = image_base64.split(",", 1)
+
+        image_bytes = base64.b64decode(encoded_image)
+
+        image = Image.open(io.BytesIO(image_bytes))
+        image_extension = image.format.lower()
+
+        # extension check
+        if image_extension not in VALID_EXTENSION:
+           raise ImageValidationError(f"invalid file extension: {image_extension}")
+
+        expected_header = f"data:image/{image_extension};base64"
+        expected_magic_header =f"image/{image_extension}"
+
+        # magic header check
+        magic_header = magic.from_buffer(image_bytes, mime=True)
+        if magic_header != expected_magic_header:
+            raise ImageValidationError(f"invalid file header: {magic_header}")
+
+        # header check
+        if header.lower() != expected_header:
+            raise ImageValidationError(f"invalid file header: {header}")
+
+        # size check
+        if image.size[0] > VALID_DIMENSION[0] and image.size[1] > VALID_DIMENSION[1]:
```

Comment:
make VALID_DIMENSION a dictionary with height and width.

Reply:
Done